File: JAVA/1572. Matrix Diagonal Sum.java
class Solution {
    public int diagonalSum(int[][] m) {
        int i,s=0;
        for(i=0;i<m.length;i++)
        {
            s+=m[i][i];
            s+=m[i][m.length-1-i];
        }
        if(m.length%2!=0)
        s-=m[m.length/2][m.length/2];
        return s;
    }
}

File: C++/1572. Matrix Diagonal Sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& m) {
        int s=0,i,j=0;
        for(i=0;i<m.size();i++){
            s+=m[i][i];
        }
        while(i!=0 && j!=m.size()){
            s+=m[--i][j++];
        }
        if(m.size()%2!=0)
        s-=m[m.size()/2][m.size()/2];
        return s;
    }
};